File: Arrays/867. Transpose Matrix/solution.py
class Solution(object):
    def transpose(self, arr):
        """
        :type matrix: List[List[int]]
        :rtype: List[List[int]]
        """
        # One liner! returns list[list[]]
        # return list(map(list, zip(*arr)))

        n, m = len(arr), len(arr[0])
        ans = [[0] * n for _ in range(m)]
        for r in range(n):
            for c in range(m):
                ans[c][r] = arr[r][c]
        return ans

File: Arrays/867. Transpose Matrix/Solution.java
class Solution {
    public int[][] transpose(int[][] matrix) {
        if (matrix.length == 0) return matrix;

        int w = matrix.length, h = matrix[0].length;
        int[][] result = new int[h][w]; // swap width and height

        for (int row = 0; row < w; row++) {
            for (int col = 0; col < h; col++) {
                result[col][row] = matrix[row][col];
            }
        }
        return result;
    }
}
